src/System.Private.ServiceModel/tools/setupfiles/sync-pr.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+show_banner()
+{
+    echo ""
+    echo "WCF Remote Github Repo sync script"
+}
+
+show_usage() 
+{
+    echo "    A URL must be specified for the pull request synchronization URL"
+    echo "    Usage: $0 [sync-url]"
+    echo "    sync-url - URL on remote server for PR synchronization"
+    echo ""
+    echo "    Example:  $0 http://wcfci-sync-server/repo/sync.ashx"
+}
+
+run_request() 
+{
+    __sync_url=$1
+    __ghprbPullId=$2
+
+    __request_uri=${__sync_url}?pr=${__ghprbPullId}
+
+    echo "Making a call to '${__request_uri}'" 
+
+    if [ ! -e "$__nugetpath" ]; then
+        which curl wget > /dev/null 2> /dev/null
+        if [ $? -ne 0 -a $? -ne 1 ]; then
+            echo "cURL or wget is required to make this request. Please see https://github.com/dotnet/wcf/blob/master/Documentation/building/unix-instructions.md for more details."
+            exit 1
+        fi
+
+        # curl has HTTPS CA trust-issues less often than wget, so lets try that first.
+        which curl > /dev/null 2> /dev/null
+        if [ $? -ne 0 ]; then
+           wget -q -O- ${__sync_url}?pr=${__ghprbPullId}
+        else
+           output=`curl -sw "\n%{http_code}" ${__sync_url}?pr=${__ghprbPullId}`
+           echo $output
+           retCode=`echo ${output} | grep "200\$"`
+        fi
+    fi
+
+    return $?
+}
+
+# Main execution
+
+show_banner
+
+# Check parameters
+
+if [ -z "$1" ]; then 
+    show_usage
+    exit 1
+fi 
+
+if [ -z "$ghprbPullId" ]; then 
+    show_usage
+    echo "This script should only be called only from the context of a GitHub Pull Request"
```

Comment:
The $2 you see above is a variable passed into the function - $1, $2... $n only mean the command line param when you're outside a function. As soon as you enter a function, then the $1, $2... $n now mean the variables that got passed into it (see line 68 on how this got called)